js_modules/dagster-ui/packages/ui-core/src/assets/catalog/useAssetCatalogGroupAndSortBy.tsx

```diff
@@ -0,0 +1,421 @@
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <AttributeStatusHeaderRow
+                {...props}
+                text={props.group}
+                groupBy={AssetHealthGroupBy.tags}
+                key={props.group}
+                noneGroup="No tags"
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.materialization_status:
+        return groupByAttribute({
+          liveDataByNode,
+          getAttributes: (asset) => {
+            return [
+              statusToIconAndColor[
+                asset.assetHealth?.materializationStatus ?? AssetHealthStatus.UNKNOWN
+              ].text,
+            ];
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <HealthStatusHeaderRow
+                {...props}
+                status={props.group}
+                groupBy={groupBy}
+                key={props.group}
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.freshness_status:
+        return groupByAttribute({
+          liveDataByNode,
+          getAttributes: (asset) => {
+            return [
+              statusToIconAndColor[asset.assetHealth?.freshnessStatus ?? AssetHealthStatus.UNKNOWN]
+                .text,
+            ];
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <HealthStatusHeaderRow
+                {...props}
+                status={props.group}
+                groupBy={groupBy}
+                key={props.group}
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.check_status:
+        return groupByAttribute({
+          liveDataByNode,
+          getAttributes: (asset) => {
+            return [
+              statusToIconAndColor[
+                asset.assetHealth?.assetChecksStatus ?? AssetHealthStatus.UNKNOWN
+              ].text,
+            ];
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <HealthStatusHeaderRow
+                {...props}
+                status={props.group}
+                groupBy={groupBy}
+                key={props.group}
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.health_status:
+      default:
+        return groupByAttribute({
+          liveDataByNode,
+          getAttributes: (asset) => {
+            return [
+              statusToIconAndColor[asset.assetHealth?.assetHealth ?? AssetHealthStatus.UNKNOWN]
+                .text,
+            ];
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <HealthStatusHeaderRow
+                {...props}
+                status={props.group}
+                groupBy={groupBy}
+                key={props.group}
+              />
+            );
+          },
+        });
+    }
+  }, [assetsByAssetKey, groupBy, liveDataByNode]);
```

Comment:
This pattern worked out great - this is long but super easy to grok and read through 💯